internal/gengapic/gengapic.go

```diff
@@ -149,30 +230,30 @@ func Gen(genReq *plugin.CodeGeneratorRequest) (*plugin.CodeGeneratorResponse, er
 		// Keep the current behavior for now, but we could revisit this later.
 		outFile := pbinfo.ReduceServName(s.GetName(), "")
 		outFile = camelToSnake(outFile)
-		outFile = filepath.Join(outDir, outFile)
+		outFile = filepath.Join(g.opts.outDir, outFile)
 
 		g.reset()
-		if err := g.gen(s, pkgName); err != nil {
+		if err := g.gen(s, g.opts.pkgName); err != nil {
 			return &g.resp, err
 		}
-		g.commit(outFile+"_client.go", pkgName)
+		g.commit(outFile+"_client.go", g.opts.pkgName)
 
 		g.reset()
-		if err := g.genExampleFile(s, pkgName); err != nil {
+		if err := g.genExampleFile(s, g.opts.pkgName); err != nil {
 			return &g.resp, errors.E(err, "example: %s", s.GetName())
 		}
-		g.imports[pbinfo.ImportSpec{Name: pkgName, Path: pkgPath}] = true
-		g.commit(outFile+"_client_example_test.go", pkgName+"_test")
+		g.imports[pbinfo.ImportSpec{Name: g.opts.pkgName, Path: g.opts.pkgPath}] = true
+		g.commit(outFile+"_client_example_test.go", g.opts.pkgName+"_test")
 	}
 
 	g.reset()
 	scopes, err := collectScopes(genServs, g.serviceConfig)
 	if err != nil {
 		return &g.resp, err
 	}
-	g.genDocFile(pkgPath, pkgName, time.Now().Year(), scopes)
+	g.genDocFile(g.opts.pkgPath, g.opts.pkgName, time.Now().Year(), scopes)
```

Comment:
Similarly here: maybe we can omit the `g.opts.xxx` args when calling the `g.yyy()` method.